LayoutTests/http/wpt/css/css-typed-om/resources/testhelper.js

```diff
@@ -0,0 +1,218 @@
+function assert_color_channel_approx_equals(a, b) {
+  // Color is is limited to 32bit RGBA, thus channels are values within 0-255.
+  // Our epsilon needs to reflect this relatively limited precision.
+  const EPSILON = 1/255;
+
+  function epsilonForUnitType(unitType) {
+    switch(unitType) {
+      case 'deg':
+        return EPSILON * 360;
+      case 'rad':
+        return EPSILON * 2 * Math.PI;
+      case 'grad':
+        return EPSILON * 400;
+      case 'percent':
+        return EPSILON * 100;
+      default:
+        return EPSILON;
+    }
+  }
+
+  assert_equals(a.constructor.name, b.constructor.name);
+  const className = a.constructor.name;
+  switch (className) {
+    case 'CSSMathSum':
+    case 'CSSMathProduct':
+    case 'CSSMathMin':
+    case 'CSSMathMax':
+      assert_equals(a.values.length, b.values.length);
+      for (let i = 0; i < a.length; i++) {
+        assert_equals(a.unit, b.unit);
+        assert_approx_equals(a[i].value, b[i].value, epsilonForUnitType(a.unit));
+      }
+      break;
+    default:
+      assert_equals(a.unit, b.unit);
+      assert_approx_equals(a.value, b.value, epsilonForUnitType(a.unit));
+  }
+}
+
+// Compares two CSSStyleValues to check if they're the same type
+// and have the same attributes.
+function assert_style_value_equals(a, b) {
+  if (a == null || b == null) {
+    assert_equals(a, b);
+    return;
+  }
+
+  assert_equals(a.constructor.name, b.constructor.name);
+  const className = a.constructor.name;
+  switch (className) {
+    case 'CSSStyleValue':
+      assert_equals(a.toString(), b.toString());
+      break;
+    case 'CSSKeywordValue':
+      assert_equals(a.value, b.value);
+      break;
+    case 'CSSUnitValue':
+      assert_approx_equals(a.value, b.value, 1e-6);
+      assert_equals(a.unit, b.unit);
+      break;
+    case 'CSSMathSum':
+    case 'CSSMathProduct':
+    case 'CSSMathMin':
+    case 'CSSMathMax':
+      assert_style_value_array_equals(a.values, b.values);
+      break;
+    case 'CSSMathClamp':
+      assert_style_value_equals(a.lower, b.lower);
+      assert_style_value_equals(a.value, b.value);
+      assert_style_value_equals(a.upper, b.upper);
+      break;
+    case 'CSSMathInvert':
+    case 'CSSMathNegate':
+      assert_style_value_equals(a.value, b.value);
+      break;
+    case 'CSSUnparsedValue':
+      assert_style_value_array_equals(a, b);
+      break;
+    case 'CSSVariableReferenceValue':
+      assert_equals(a.variable, b.variable);
+      assert_style_value_equals(a.fallback, b.fallback);
+      break;
+    case 'CSSPositionValue':
```

Comment:
`CSSPositionValue` is not a thing anymore. Is this being imported from the WPT repo?